CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,503 @@
+              TupleExprElementSyntax(expression: ExprSyntax("(SyntaxArena(), (\(parameters)))"))
+            },
+            rightParen: .rightParenToken(),
+            trailingClosure: ClosureExprSyntax(signature: closureSignature) {
+              if node.children.isEmpty {
+                DeclSyntax("let raw = RawSyntax.makeEmptyLayout(kind: SyntaxKind.\(raw: node.swiftSyntaxKind), arena: arena)")
+              } else {
+                DeclSyntax("let layout: [RawSyntax?] = \(layoutList)")
+                DeclSyntax(
+                  """
+                  let raw = RawSyntax.makeLayout(
+                    kind: SyntaxKind.\(raw: node.swiftSyntaxKind), from: layout, arena: arena,
+                    leadingTrivia: leadingTrivia, trailingTrivia: trailingTrivia)
+                  """
+                )
+              }
+              StmtSyntax("return SyntaxData.forRoot(raw)")
+            }
+          )
+
+          VariableDeclSyntax(
+            leadingTrivia: """
+              // Extend the lifetime of all parameters so their arenas don't get destroyed
+              // before they can be added as children of the new arena.
+
+              """,
+            .let,
+            name: PatternSyntax("data"),
+            type: TypeAnnotationSyntax(type: TypeSyntax("SyntaxData")),
+            initializer: InitializerClauseSyntax(value: initializer)
+          )
+
+          ExprSyntax("self.init(data)")
+        }
+
+        if node.hasOptionalBaseTypeChild {
+          // TODO: Remove when we no longer support compiling in Swift 5.6. Change the
+          // above constructor to use `Optional<BaseType>.none` instead.
+          try InitializerDeclSyntax(
+            """
+            /// This initializer exists solely because Swift 5.6 does not support
+            /// `Optional<ConcreteType>.none` as a default value of a generic parameter.
+            /// The above initializer thus defaults to `nil` instead, but that means it
+            /// is not actually callable when either not passing the defaulted parameter,
+            /// or passing `nil`.
+            ///
+            /// Hack around that limitation using this initializer, which takes a
+            /// `Missing*` syntax node instead. `Missing*` is used over the base type as
+            /// the base type would allow implicit conversion from a string literal,
+            /// which the above initializer doesn't support.
+            \(node.generateInitializerDeclHeader(optionalBaseAsMissing: true))
+            """
+          ) {
+            FunctionCallExprSyntax(callee: ExprSyntax("self.init")) {
+              TupleExprElementSyntax(label: "leadingTrivia", expression: ExprSyntax("leadingTrivia"))
+              for child in node.children {
+                if child.hasOptionalBaseType {
+                  TupleExprElementSyntax(label: child.swiftName, expression: ExprSyntax("Optional<\(raw: child.typeName)>.none"))
```

Comment:
I think the generated source could would be a fair amount more readable if all these parameters are on separate lines. Should be doable by just adding a leading newline to all the `TupleExprElementSyntax` here.

Reply:
Also added newline in the init signature right above